contrib/ubase/patches/su.patch

```diff
@@ -0,0 +1,42 @@
+Index: ubase/ubase/su.c
+===================================================================
+--- ubase.orig/ubase/su.c	2023-05-30 17:51:24.363395446 +0200
++++ ubase/ubase/su.c	2023-05-30 17:55:05.003401122 +0200
+@@ -14,6 +14,7 @@
+ #include "util.h"
+ 
+ extern char **environ;
++extern char *envinit[1];
+ 
+ static int lflag = 0;
+ static int pflag = 0;
+@@ -21,9 +22,9 @@
+ static int
+ dologin(struct passwd *pw)
+ {
+-	char *shell = pw->pw_shell[0] == '\0' ? "/bin/sh" : pw->pw_shell;
++	const char *shell = pw->pw_shell[0] == '\0' ? "/bin/sh" : pw->pw_shell;
+ 	char *term = getenv("TERM");
+-	clearenv();
++	environ = envinit;
```

Comment:
What about defining `clearenv` in `util.h` and adding it to `libutil/`?

Reply:
I'm not sure in which file in `ubase/libutil/` should I implement this function. 
I tried to create a new separate `ubase/libutil/clearenv.c` file for it by adding a new `quilt` patch and add it to `LIBUTILSRC` in https://github.com/panantoni01/mimiker/blob/ubase/contrib/ubase/Makefile#L22. However with this approach the new source file cannot compiled during the build process, because quilt patches for ubase are applied **after** generating `*.d` files for the ubase object files.

Reply:
Ouch... maybe implement it as `static inline` in `util.h` then? Since the functionality is used more than once I'd like to keep it in a single place.

Reply:
Added `clearenv` as `static inline` in a separate patch for `util.h` in https://github.com/cahirwpz/mimiker/pull/1371/commits/122906fd65f14419f8638a3346bd2dc9ca12f5d6